system-contracts/contracts/precompiles/EcMul.yul

```diff
@@ -72,423 +54,35 @@ object "EcMul" {
-            /// @param xp The x coordinate of the point P in projective coordinates in Montgomery form.
-            /// @param yp The y coordinate of the point P in projective coordinates in Montgomery form.
-            /// @param zp The z coordinate of the point P in projective coordinates in Montgomery form.
-            /// @return xr The x coordinate of the point P in affine coordinates in Montgomery form.
-            /// @return yr The y coordinate of the point P in affine coordinates in Montgomery form.
-            function projectiveIntoAffine(xp, yp, zp) -> xr, yr {
-                if zp {
-                     let zp_inv := montgomeryModularInverse(zp)
-                     xr := montgomeryMul(xp, zp_inv)
-                     yr := montgomeryMul(yp, zp_inv)
-                 }
-            }
-
-            /// @notice Doubles a point in projective coordinates in Montgomery form.
-            /// @dev See Algorithm 9 in https://eprint.iacr.org/2015/1060.pdf for further details.
-            /// @dev The point is assumed to be on the curve.
-            /// @dev It works correctly for the point at infinity.
-            /// @param xp The x coordinate of the point P in projective coordinates in Montgomery form.
-            /// @param yp The y coordinate of the point P in projective coordinates in Montgomery form.
-            /// @param zp The z coordinate of the point P in projective coordinates in Montgomery form.
-            /// @return xr The x coordinate of the point 2P in projective coordinates in Montgomery form.
-            /// @return yr The y coordinate of the point 2P in projective coordinates in Montgomery form.
-            /// @return zr The z coordinate of the point 2P in projective coordinates in Montgomery form.
-            function projectiveDouble(xp, yp, zp) -> xr, yr, zr {
-                let t0 := montgomeryMul(yp, yp)
-                zr := montgomeryAdd(t0, t0)
-                zr := montgomeryAdd(zr, zr)
-                zr := montgomeryAdd(zr, zr)
-                let t1 := montgomeryMul(yp, zp)
-                let t2 := montgomeryMul(zp, zp)
-                t2 := montgomeryMul(MONTGOMERY_B3(), t2)
-                xr := montgomeryMul(t2, zr)
-                yr := montgomeryAdd(t0, t2)
-                zr := montgomeryMul(t1, zr)
-                t1 := montgomeryAdd(t2, t2)
-                t2 := montgomeryAdd(t1, t2)
-                t0 := montgomerySub(t0, t2)
-                yr := montgomeryMul(t0, yr)
-                yr := montgomeryAdd(xr, yr)
-                t1 := montgomeryMul(xp, yp)
-                xr := montgomeryMul(t0, t1)
-                xr := montgomeryAdd(xr, xr)
-            }
-
             ////////////////////////////////////////////////////////////////
             //                      FALLBACK
             ////////////////////////////////////////////////////////////////
 
             // Retrieve the coordinates from the calldata
             let x := calldataload(0)
```

Comment:
The same here, you can do `calldatacopy` instead as there are no checks in the contract